.dev/base-files/README.md

```diff
@@ -0,0 +1,258 @@
+test your app against a local JupyterHub instance.
+
+Run ``python -m ci preflight run`` to run JupyterHub.  You can log in at
+http://127.0.0.1:8000.  Enter the username "edge" and the password "password".
+When finished, type Ctrl-C to exit.
+
+To run the preflight check in an automated way, do 
+``python -m ci preflight test``.  This can be e.g. incorporated into a 
+GitHub Actions workflow.
+
+
+## edge_settings.json
+
+You will notice an edge_settings.json.example file in this example's root.
+During local development, it is often convenient to have a genuine EdgeSession
+object for e.g. interacting with the Edge file system.  To make an EdgeSession
+available in your app during development, fill in the fields in this file
+and rename it to "edge_settings.json".  The "ci" module will pick up the
+information and set the appropriate environment variables when your app runs.
+
+**Please be careful not to check edge_settings.json into source control, and
+do not copy it into your Docker image.**  When running in the real Edge
+system, the EdgeSession object can be created automatically using the
+token, etc., of the user who launched the app.  This information is scraped
+from the environment when you call the ``EdgeSession()`` constructor.
+
+
+## Environment variables
+
+Within your app, you have access to a number of environment variables which
+may be useful for the development process:
+
+APP_VERSION: In local mode, this is a string with the Docker image name + tag.
+**When running in the real Edge system, this variable will not be defined.**
+
+NATIVE_APP_MODE: When running with ``python -m ci dev``, this will be set to
+"dev".  When running with ``python -m ci container``, this will be set to
+"container".  **When running in the real Edge system, this variable
+will not be defined.**  In general, you should try to minimize logic that depends on
+this variable, to avoid unexpected behavior when moving from local development
+to production.
+
+
+## Managing EDM and Pip dependencies
+
+In config.py, you can specify both _local development_ dependencies, which
+will be used in your bootstrap.py-built development environment, and
+_application dependencies_, which will be installed into your Docker image.
+
+By default, both will use the list of Deployment Server repositories given
+in your ``~/edm.yaml`` file, along with any credentials in that file.  You
+can select a different ``edm.yaml`` file using the ``--edm-config`` option
+to commands like ``python -m ci container build``.
+
+If you need to install packages from Pip, you can do so in config.py for
+local development dependencies.  For application dependencies, you should
+put the Pip command in your Dockerfile.  We recommend something like this:
+```
+RUN edm run -e <env name> -- pip install --no-cache-dir <requirements>
+```
+Using ``--no-cache-dir`` will reduce the size of the Docker image.
+
+
+## Registering versions of your app with Edge
```

Comment:
In the future should this be promoted to being a CI tool?

Reply:
Possibly, although I would like to encourage people to use the Edge API directly for now.